Sources/Foundation/JSONSerialization.swift

```diff
@@ -306,7 +309,7 @@ open class JSONSerialization : NSObject {
 
 private extension JSONSerialization {
     /// Detect the encoding format of the NSData contents
-    static func detectEncoding(_ bytes: UnsafeRawBufferPointer) -> (String.Encoding, Int) {
+    static func detectEncoding(_ bytes: Data) -> (String.Encoding, Int) {
```

Comment:
The `bytes[0]`, etc. subscripts in this function are subtly incorrect for `Data`. A `Data` happens to be its own slice type, and if it's a slice, it can have a non-zero `startIndex`, which will result in a precondition failure for `bytes[0]`, etc.

Reply:
(This is handled properly in the `DocumentReader` implementation, which performs access bounded by `startIndex` and `endIndex`.)